[4.6] chore(a11y): generate wai aria articles with the latest a11y specs (#1666)
telerik/blazor-docs: components/pdfviewer/accessibility/wai-aria-support.md

---
title: Wai-Aria Support
page_title: Telerik UI for Blazor PDFViewer Documentation | PDFViewer  Accessibility
description: "Get started with the Telerik UI for Blazor PDFViewer and learn about its accessibility support for WAI-ARIA, Section 508, and WCAG 2.1."
tags: telerik,blazor,accessibility,wai-aria,wcag
slug: pdfviewer-wai-aria-support 
position: 50 
---

# Blazor PDFViewer Accessibility

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

> 
> 

Reviewer: ```suggestion
```

> 
> 

Reviewer: ```suggestion
```

Out of the box, the Telerik UI for Blazor PDFViewer provides extensive accessibility support and enables users with disabilities to acquire complete control over its features.

> The PDFViewer is compliant with the [Web Content Accessibility Guidelines (WCAG) 2.1  AA](https://www.w3.org/TR/WCAG21/) standards](https://www.w3.org/TR/WCAG21/) and [Section 508](http://www.section508.gov/) requirements, follows the [Web Accessibility Initiative - Accessible Rich Internet Applications (WAI-ARIA)](https://www.w3.org/WAI/ARIA/apg/) best practices for implementing the [keyboard navigation](#keyboard-navigation) for its `component` role, provides options for managing its focus and is tested against the most popular screen readers.

Reviewer: Split to a list for better readability.

## WAI-ARIA

> 

Reviewer: ```suggestion
```

This section lists the selectors, attributes, and behavior patterns supported by the component and its composite elements, if any.

> 

Reviewer: ```suggestion
```

> The PDFViewer component contains two inner elements - a toolbar and a page container.

Reviewer 2: ```suggestion
The PDFViewer component contains two inner elements—a toolbar and a page container.
```

Author: This PR only triggers the autogeneration of this article. Your suggestion will be taken into account after we merge this.

> [ToolBar accessibility specification]({{toolbar_a11y_link}})

Reviewer 2: Is this link related to the table below? I'm struggling to understand why it's placed like this between the previous sentence and the table below. Consider using this anchor as a part of a complete sentence.

Author: This PR only triggers the autogeneration of this article and causes placeholders in some places. It will be fixed.

| Selector | Attribute | Usage |
| -------- | --------- | ----- |
| `.k-pdfviewer .k-canvas` | `tabindex=0` | Defines the focusable page container element. |
|  | `aria-label` | Describes the purpose of the focusable container. Translatable message. |
|  | `role=document` | Defines that content should be evaluated in reader mode by assistive technologies. |
| `.k-pdfviewer .k-toolbar .k-button:has(.k-svg-i-search, .k-i-search)` | `aria-haspopup=dialog` | Describes that the Search tool button opens a dialog element. |
| `.k-pdfviewer .k-canvas .k-search-panel` | `role=dialog` | Describes the role of the Search panel. |
|  | `aria-label` | Translatable message, same label as the one, used to describe the Toolbar Search tool. |

## Section 508

> 

Reviewer: ```suggestion
```

The PDFViewer is fully compliant with the [Section 508 requirements](http://www.section508.gov/).

## Testing

> 

Reviewer: ```suggestion
```

The PDFViewer has been extensively tested automatically with [axe-core](https://github.com/dequelabs/axe-core) and manually with the most popular screen readers.

> To report any accessibility issues, contact the team through the [Telerik Support System](https://www.telerik.com/account/support-center).

### Screen Readers

> 

Reviewer: ```suggestion
```

> The PDFViewer has been tested with the following screen readers and browsers combinations:

Reviewer: When a noun is used as an adjective, it must be in singular.

```suggestion
The PDFViewer has been tested with the following screen reader and browser combinations:
```

> | Environment | Tool |

Reviewer: ```suggestion
| Browser | Screen Reader |
```

| ----------- | ---- |
| Firefox | NVDA |
| Chrome | JAWS |
| Microsoft Edge | JAWS |

> 

Reviewer: ```suggestion
```
